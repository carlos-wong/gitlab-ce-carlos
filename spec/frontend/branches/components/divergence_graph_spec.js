import { shallowMount } from '@vue/test-utils';
import DivergenceGraph from '~/branches/components/divergence_graph.vue';
import GraphBar from '~/branches/components/graph_bar.vue';

let vm;

function factory(propsData = {}) {
  vm = shallowMount(DivergenceGraph, { propsData });
}

describe('Branch divergence graph component', () => {
  afterEach(() => {
    vm.destroy();
  });

  it('renders ahead and behind count', () => {
    factory({
      defaultBranch: 'master',
      aheadCount: 10,
      behindCount: 10,
      maxCommits: 100,
    });

    expect(vm.findAll(GraphBar).length).toBe(2);
    expect(vm.element).toMatchSnapshot();
  });

  it('sets title for ahead and behind count', () => {
    factory({
      defaultBranch: 'master',
      aheadCount: 10,
      behindCount: 10,
      maxCommits: 100,
    });

    expect(vm.attributes('title')).toBe('10 commits behind master, 10 commits ahead');
  });

  it('renders distance count', () => {
    factory({
      defaultBranch: 'master',
      aheadCount: 0,
      behindCount: 0,
      distance: 900,
      maxCommits: 100,
    });

    expect(vm.findAll(GraphBar).length).toBe(1);
    expect(vm.element).toMatchSnapshot();
  });

  it.each`
    distance | titleText
    ${900}   | ${'900'}
    ${1100}  | ${'999+'}
  `('sets title for $distance as $titleText', ({ distance, titleText }) => {
    factory({
      defaultBranch: 'master',
      aheadCount: 0,
      behindCount: 0,
      distance,
      maxCommits: 100,
    });

    expect(vm.attributes('title')).toBe(`More than ${titleText} commits different with master`);
  });
});
