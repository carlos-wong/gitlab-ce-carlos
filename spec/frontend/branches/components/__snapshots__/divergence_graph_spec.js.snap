// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Branch divergence graph component renders ahead and behind count 1`] = `
<div
  class="divergence-graph px-2 d-none d-md-block"
  title="10 commits behind master, 10 commits ahead"
>
  <graph-bar-stub
    count="10"
    maxcommits="100"
    position="left"
  />
   
  <div
    class="graph-separator pull-left mt-1"
  />
   
  <graph-bar-stub
    count="10"
    maxcommits="100"
    position="right"
  />
</div>
`;

exports[`Branch divergence graph component renders distance count 1`] = `
<div
  class="divergence-graph px-2 d-none d-md-block"
  title="More than 900 commits different with master"
>
  <graph-bar-stub
    count="900"
    maxcommits="100"
    position="full"
  />
</div>
`;
